genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/FinishedJob.java

```diff
@@ -0,0 +1,361 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.common.internal.dto.v4;
+
+import com.google.common.collect.ImmutableList;
+import com.netflix.genie.common.dto.JobStatus;
+import com.netflix.genie.common.internal.exceptions.unchecked.GenieInvalidStatusException;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * DTO for a job that reached a final state.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Getter
+@ToString(callSuper = true, doNotUseGetters = true)
+@EqualsAndHashCode(callSuper = true, doNotUseGetters = true)
+@SuppressWarnings("checkstyle:finalclass")
+public class FinishedJob extends CommonMetadata {
+
+    private final String uniqueId;
+    private final Instant created;
+    private final JobStatus status;
+    private final List<String> commandArgs;
+    private final Criterion commandCriterion;
+    private final List<Criterion> clusterCriteria;
+    private final Optional<Instant> started;
```

Comment:
I understand this is to avoid writing all these getters/boilerplate but geez it's ugly/not consistent with the rest of our classes